webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/services/LogDisplayer.scala

```diff
@@ -92,18 +93,36 @@ class LogDisplayer(
       (for {
         parsed   <- tryo(parse(jsonInterval)) ?~! s"Error when trying to parse '${jsonInterval}' as a JSON datastructure with fields 'start' and 'end'"
         startStr <- parsed \ "start" match {
-                      case JString(startStr) => tryo(DateTime.parse(startStr, format)) ?~! s"Bad format for start date, was execpting '${format.toString}' and got '${startStr}'"
-                      case x => Failure("Error: missing start date and time")
-                    }
+          case JString(startStr) if startStr.nonEmpty =>
+            val date = tryo(DateTime.parse(startStr, format)) ?~! s"Error when trying to parse start date '${startStr}"
+            date match {
+              case Full(d) => Full(Some(d))
+              case eb: EmptyBox =>
+                eb ?~! s"Invalid start date"
+            }
+          case _ => Full(None)
+        }
         endStr   <- parsed \ "end" match {
-                      case JString(endStr) => tryo(DateTime.parse(endStr, format)) ?~! s"Bad format for end date, was execpting '${format.toString}' and got '${endStr}'"
-                      case x => Failure("Error: missing end date and time")
-                    }
+          case JString(endStr) if endStr.nonEmpty =>
+            val date = tryo(DateTime.parse(endStr, format)) ?~! s"Error when trying to parse end date '${endStr}"
+            date match {
+              case Full(d) => Full(Some(d))
+              case eb: EmptyBox =>
+                eb ?~! s"Invalid end date"
+            }
+          case _ => Full(None)
+        }
       } yield {
-        if(startStr.isBefore(endStr)) {
-          reportRepository.findReportsByNodeOnInterval(nodeId, startStr, endStr)
-        } else {
-          reportRepository.findReportsByNodeOnInterval(nodeId, endStr, startStr)
+        (startStr, endStr) match {
+          case (None, None) =>
+            reportRepository.findReportsByNodeOnInterval(nodeId, new DateTime(0), new DateTime())
```

Comment:
You may better modify findReportByNodeOnInterval to take options so you don't have to make DateTime(0) and Datetime.now (which you should use instead of new DateTime())

Reply:
This would result in a two case pattern matching 

(start,end) match {
  case (Some(startValue),Some(endValue)) if startValue.isAfter(endValue) =>
    reportRepository.findReportsByNodeOnInterval(nodeId, end, start)
  case _ =>
    reportRepository.findReportsByNodeOnInterval(nodeId, start, end)
}

Reply:
Should I use ` toQuery0((nodeId, start.getOrElse(new DateTime(0)), end.getOrElse(DateTime.now)))` or there is an other way to do it ?